HM3CoreKill/HM3CoreKill/ck/HM3Hooks.cpp

```diff
@@ -148,4 +157,86 @@ void __stdcall ZHM3_OnAnimationLoaded(ioi::hm3::ZAnimationInfo* animationInstanc
 	{		
 		ck::HM3AnimationRegistry::getRegistry().registerAnimation(animationInstance);
 	}
+}
+
+void __stdcall CMapObject_OnCreate(ioi::hm3::CMapObject* instance)
+{
+	if (!instance)
+		return;
+
+	HM3_TRACE_NATIVE_OBJECT_CREATION(instance);
+}
+
+void __stdcall ZGlacier_OnSTDOBJAttached(DWORD* unknownInstance)
+{
+	if (!unknownInstance)
+		return;
+
+	ioi::hm3::ZGlacierRTTI* runtimeInfo = ioi::hm3::getTypeInfo(unknownInstance);
+	if (!runtimeInfo)
+		return;
+
+	const std::string_view parentType = runtimeInfo->Parent;
+
+	if (parentType == "ZGEOM")
+	{
+		const std::string_view instanceType = runtimeInfo->SelfType;
+		const bool isTypeBasedOnDoorWord = instanceType.find("Door", 0) != std::string_view::npos;
+		if (isTypeBasedOnDoorWord)
+		{
+			ioi::hm3::CDoor* instance = reinterpret_cast<ioi::hm3::CDoor*>(unknownInstance);
+			ck::HM3DoorsRegistry::getRegistry().registerDoor(instance);
+			return;
+		}
+	}
+}
+
+struct FsZip_Hooked
+{
+	int readHooked(const char* name, void* dest, int unk0, int unk1)
+	{
+		typedef int(__thiscall* FsZip_read_t)(void*, const char*, void*, int, int);
+
+		const int LevelIdOffset = 7;
+		const int LevelIdLength = 3;
+
+		char levelID[LevelIdLength + 1] = { 0 };
+		memcpy((void*)&levelID, name + LevelIdOffset, LevelIdLength);
+
+		std::string_view path(name);
+
+		/// TODO: Fix this dirty code!
+
+		if (name && name[0] != '*' && path.find("Scenes\\") == 0)
```

Comment:
📛 This is test code. Need to write smart router for better file location process.